ordinary_data/valve/tr_valve_pipe.sql

```diff
@@ -5,7 +5,49 @@
 */
 
 /* UPDATES NUMBER OF VALVES AND CLOSED/OPEN FOR PIPES */
-CREATE OR REPLACE FUNCTION qwat_od.fn_pipe_update_valve(_pipe_id integer) RETURNS VOID AS
+CREATE OR REPLACE FUNCTION qwat_od.fn_pipe_delete_valve(_valve_id integer)
+  RETURNS void AS
+$BODY$
+BEGIN
+
+IF (SELECT _valve_count > 1 FROM qwat_od.pipe WHERE id = (
```

Comment:
would it do any harm to skip this check?
if the field is not correct, it might be good to go further.
And I don't think this is too costless, is it?

Reply:
Without this test, you can let old count not modified.